server/mediaserver.go

```diff
@@ -510,10 +511,88 @@ func getRTMPStreamHandler(s *LivepeerServer) func(url *url.URL) (stream.RTMPVide
 
 //End RTMP Handlers
 
+func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
+
+	// we read this unconditionally, mostly for ffmpeg
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		glog.Info(err)
+		return
+	}
+	r.Body.Close()
+
+	if ".ts" != path.Ext(r.URL.Path) {
+		// ffmpeg sends us a m3u8 as well, so ignore
+		// Alternatively, reject m3u8s explicitly and take any other type
+		// TODO also look at use content-type
+		return
+	}
+
+	mid := parseManifestID(r.URL.Path)
+	s.connectionLock.RLock()
+	cxn, exists := s.rtmpConnections[mid]
+	s.connectionLock.RUnlock()
+
+	// Check for presence and register if a fresh cxn
+	if !exists {
+		appData := (createRTMPStreamIDHandler(s))(r.URL)
```

Comment:
This (I think this) leads to a slight inconsistency in the webhook handling between HTTP and RTMP — the RTMP URLs come in with the full URL protocol like `rtmp://example.com/streamkey` but HTTP hooks come in with just the path `/live/streamkey/0.ts`. It'd probably be good to include the protocol in all cases in case the webhook server wants to disallow a protocol or something.

Reply:
Ah yeah looks like the `r.URL` here is simply extracted from the R-URI field and that's usually missing the scheme, host, port, etc. Looks like that'll need to be reconstructed from the golang [Request](https://golang.org/pkg/net/http/#Request) object.

Reply:
Updated the writeup to mention this should be addressed. Thanks!

Reply:
```r.URL.RequestURI()``` ?